chalk-solve/src/solve/slg.rs

```diff
@@ -317,6 +322,142 @@ fn into_ex_clause(result: UnificationResult, ex_clause: &mut ExClause<SlgContext
     ex_clause.constraints.extend(result.constraints);
 }
 
+/// Returns a set of program clauses that could possibly match
+/// `goal`. This can be any superset of the correct set, but the
+/// more precise you can make it, the more efficient solving will
+/// be.
+fn program_clauses_that_could_match(
```

Comment:
Nit: I think this logic should live in the `crate::clauses` module, actually. I suspect this is the only real consumer of the `ToProgramClauses` trait.